client/test/unit/specs/transactions.spec.js

```diff
@@ -1,92 +1,164 @@
-import Vue from 'vue'
-import VDragged from 'v-dragged'
+import Vue from 'vue';
+import VDragged from 'v-dragged';
 
-import Transactions from '@/components/application/Transactions'
-import store from '@/store'
-import { cleanState } from '../../features/specs/support/clean.state'
-import sinon from 'sinon'
-import axios from '@/axios-auth'
+import Transactions from '@/components/application/Transactions';
+import store from '@/store';
+import { cleanState } from '../../features/specs/support/clean.state';
 
-Vue.use(VDragged)
+Vue.use(VDragged);
 
 describe('Testing Transactions.vue', () => {
-  let data = {}
-  let sandbox
-  const Constructor = Vue.extend(Transactions)
+  const data = {};
+  let sandbox;
+  const Constructor = Vue.extend(Transactions);
 
-  beforeAll(done => {
-    sandbox = sinon.createSandbox()
-    sandbox.stub(axios, 'get').resolves(
-      {
-        data:
-          {
-            response: { count: 3 },
-            transactions:
-               { 1:
-                  { action: 'update_from_nro',
-                    eventDate: 'Wed, 29 May 2019 19:53:01 GMT',
-                    id: 1998450,
-                    jsonData: {},
-                    requestId: 2257876,
-                    stateCd: 'INPROGRESS',
-                    userId: 80,
-                    user_action: 'Get NR Details from NRO',
-                    user_name: 'github/scottrumsby' },
-                 2:
-                  { action: 'get',
-                    eventDate: 'Wed, 29 May 2019 19:53:01 GMT',
-                    id: 1998451,
-                    jsonData: {},
-                    requestId: 2257876,
-                    stateCd: 'INPROGRESS',
-                    userId: 80,
-                    user_action: 'Get Next NR',
-                    user_name: 'github/scottrumsby' },
-                 3:
-                  { action: 'marked_on_hold',
-                    eventDate: 'Wed, 29 May 2019 20:30:41 GMT',
-                    id: 1998452,
-                    jsonData: {},
-                    requestId: 2257876,
-                    stateCd: 'HOLD',
-                    userId: 1,
-                    user_action: '',
-                    user_name: 'nro_service_account'
-                  }
-              }
-          }
-      })
-    store.replaceState(cleanState())
-    data.instance = new Constructor({ store })
-    data.vm = data.instance.$mount()
-    data.vm.$store.dispatch('getTransactionsHistory')
-    data.vm.$store.state.transactionsModalVisible = true
+  beforeAll((done) => {
+    store.replaceState(cleanState());
+    data.instance = new Constructor({ store });
+    data.vm = data.instance.$mount();
+    data.vm.$store.dispatch('getTransactionsHistory');
+    data.vm.$store.state.transactionsModalVisible = true;
     setTimeout(() => {
-      done()
-    }, 1000)
-  })
+      done();
+    }, 1000);
+  });
+
+  beforeEach(() => {
+    data.vm.$store.state.transactionsData = [
```

Comment:
Well since you've refactored this... the transaction data structure mostly doesn't match what comes from the API.

Here is what I started:
```
transactions: [
  {
    additionalInfo: "*** New Request ***",
    consentFlag: "N",
    consent_dt: null,
    corpNum: null,
    eventDate: "Mon, 12 Jul 2021 17:20:20 GMT",
    expirationDate: null,
    furnished: "N",
    names: [
      {
        choice: 1,
        comment: null,
        conflict1: "",
        conflict1_num: "",
        conflict2: "",
        conflict2_num: "",
        conflict3: "",
        conflict3_num: "",
        consumptionDate: null,
        corpNum: null,
        decision_text: "",
        designation: "",
        id: 4263914,
        name: "CODE REVIEW",
        name_type_cd: "CO",
        state: "NE"
      },
      {
        choice: 2,
        comment: null,
        conflict1: "",
        conflict1_num: "",
        conflict2: "",
        conflict2_num: "",
        conflict3: "",
        conflict3_num: "",
        consumptionDate: null,
        corpNum: null,
        decision_text: "",
        designation: "",
        id: 4263915,
        name: "REVIEW CODE",
        name_type_cd: "CO",
        state: "NE"
      }
    ],
    priorityCd: "Y",
    requestTypeCd: "XUL",
    request_action_cd: "NEW",
    stateCd: "APPROVED",
    user_action: "Decision",
    user_name: "sriyazzu@idir"
  },
  {
    additionalInfo: "*** New Request ***",
    consentFlag: "N",
    consent_dt: null,
    corpNum: "",
    eventDate: "Mon, 12 Jul 2021 16:50:45 GMT",
    expirationDate: null,
    furnished: "N",
    names: [
      {
        choice: 1,
        comment: null,
        conflict1: "",
        conflict1_num: "",
        conflict2: "",
        conflict2_num: "",
        conflict3: "",
        conflict3_num: "",
        consumptionDate: null,
        corpNum: null,
        decision_text: "",
        designation: "",
        id: 4263914,
        name: "CODE REVIEW",
        name_type_cd: "CO",
        state: "NE"
      },
      {
        choice: 2,
        comment: null,
        conflict1: "",
        conflict1_num: "",
        conflict2: "",
        conflict2_num: "",
        conflict3: "",
        conflict3_num: "",
        consumptionDate: null,
        corpNum: null,
        decision_text: "",
        designation: "",
        id: 4263915,
        name: "REVIEW CODE",
        name_type_cd: "CO",
        state: "NE"
      }
    ],
    priorityCd: "Y",
    requestTypeCd: "XUL",
    request_action_cd: "NEW",
    stateCd: "INPROGRESS",
    user_action: "Load NR",
    user_name: "sriyazzu@idir"
  },
  {
    additionalInfo: "*** New Request ***",
    consentFlag: null,
    consent_dt: null,
    corpNum: "",
    eventDate: "Mon, 12 Jul 2021 16:49:50 GMT",
    expirationDate: null,
    furnished: "N",
    names: [
      {
        choice: 1,
        comment: null,
        conflict1: "",
        conflict1_num: "",
        conflict2: "",
        conflict2_num: "",
        conflict3: "",
        conflict3_num: "",
        consumptionDate: null,
        corpNum: null,
        decision_text: "",
        designation: "",
        id: 4263914,
        name: "CODE REVIEW",
        name_type_cd: "CO",
        state: "NE"
      },
      {
        choice: 2,
        comment: null,
        conflict1: "",
        conflict1_num: "",
        conflict2: "",
        conflict2_num: "",
        conflict3: "",
        conflict3_num: "",
        consumptionDate: null,
        corpNum: null,
        decision_text: "",
        designation: "",
        id: 4263915,
        name: "REVIEW CODE",
        name_type_cd: "CO",
        state: "NE"
      }
    ],
    priorityCd: "Y",
    requestTypeCd: "XUL",
    request_action_cd: "NEW",
    stateCd: "DRAFT",
    user_action: "Created NR (Payment Completed)",
    user_name: "name_request_service_account"
  }
]
```
You should refactor this to match the correct data structure. Don't worry about my data -- I will build upon whatever you implement.

Reply:
Yes you are right, i reused what there was with modified properties which made no sense. 
I've updated the tests to actually represent the structure of current transactions in next commit.